From 9f848516663fd244eb6694123faa2fd5e6350ab5 Mon Sep 17 00:00:00 2001
From: Loic Poulain <loic.poulain@intel.com>
Date: Wed, 1 Oct 2014 16:49:14 +0200
Subject: [PATCH 5/5] net: rfkill: gpio: Add BCM47521 id

Add BCM47521 to the ACPI table.
BCM47521 supports host wake.
HOST <---UART---> CONTROLLER
HOST <-- WAKE---- CONTROLLER (gpio res 0)
HOST ---ENABLE--> CONTROLLER (gpio res 1)

Signed-off-by: Loic Poulain <loic.poulain@intel.com>
---
 net/rfkill/rfkill-gpio.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index a4c4826..8090423 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -321,6 +321,14 @@ static struct rfkill_gpio_desc acpi_bluetooth_wake = {
 	.host_wake_idx = 2,
 };
 
+static struct rfkill_gpio_desc acpi_gps_wake = {
+	.type = RFKILL_TYPE_GPS,
+	.reset_idx = -1,
+	.shutdown_idx = 1,
+	.wake_idx = -1,
+	.host_wake_idx = 0,
+};
+
 static const struct acpi_device_id rfkill_acpi_match[] = {
 	{ "BCM2E1A", (kernel_ulong_t)&acpi_default_bluetooth },
 	{ "BCM2E39", (kernel_ulong_t)&acpi_default_bluetooth },
@@ -329,6 +337,7 @@ static const struct acpi_device_id rfkill_acpi_match[] = {
 	{ "BCM4752", (kernel_ulong_t)&acpi_default_gps },
 	{ "LNV4752", (kernel_ulong_t)&acpi_default_gps },
 	{ "BCM4752E", (kernel_ulong_t)&acpi_default_gps },
+	{ "BCM47521", (kernel_ulong_t)&acpi_gps_wake },
 	{ },
 };
 MODULE_DEVICE_TABLE(acpi, rfkill_acpi_match);
-- 
1.8.3.2

